### ISSUE TITLE
Unable to get wwsympa to run on Debian

### ISSUE THREAD

**Issue by qulien** [sympa-community/sympa#1256]
I've installed sympa (and with it, wwsympa) on Debian Bullseye, as well as Bookworm. In both cases, I was unable to get the web frontend (wwsympa) to work. I've tried _a lot_ of things, have probably spent about 20 hours trying to get it to work. At this time I would like to ask the community for some support, if possible.

- Server has a domain name, let's say `sympa.redacted.com`
- It is supposed to only run sympa, so `sympa.redacted.com` is supposed to show wwsympa
- I want to use an external postfix server, so the mail server setup hasn't been performed yet (that will be my next challenge)

One interesting observation I made is that the auto-generated (by the installer) Apache configuration (`/etc/apache2/conf-available/sympa.conf`) has the following line:

    ScriptAlias /wws /usr/lib/cgi-bin/sympa/wwsympa-wrapper.fcgi

However, there is no such file in that directory. Might this be a packaging bug, I wonder?

Also, I wonder, am I supposed to create a `sympa.conf` in `/etc/apache2/sites-available/` myself, or should this be taken care of by the installer or should this not be necessary? If I am supposed to create it (I assume I am), then should I include the config from within this file?

Those are just a few of the dozens of questions I have. I'm afraid I could really do with someone holding my hand and walking me through the setup step-by-step. I would assume a lot of users run this on Debian, so I was astonished that I couldn't find anything about the issues I'm experiencing. Maybe I'm doing something fundamentally wrong.

Version
-------
- Sympa 6.2.60 on Debian 11
- Sympa 6.2.60 on Debian 10

Installation method
-------------------
Via `apt install sympa` and then following up with some steps from the Sympa manual.

Expected behavior
-----------------
The web interface should be available at `sympa.redacted.com/wws` after installation.

Actual behavior
---------------
The web interface isn't available. I get 404 not found. After creating a `sympa.conf` in `sites-available` and testing different configurations and options, I get varying results, but never a working web interface.

Additional information
----------------------
None.

**Comment by qosobrin** [comment]
Any advance in this topic for Debian Bullseye, @racke? I was trying to make it work and it was driving me mad. Could anyone provide detailed instructions on how to make the web interface of sympa work in this version of Debian?

Meanwhile, I came up with a solution that worked for me but I do not know if it is the best. I had previously used this solution to install a testing environment for sympa from source files (issue #1283) so I adapted it for a distribution installation. Here it is, @qulien, for your information but take it with caution since it is not fully tested (or tested at all more than opening the web page :

1. I started installing this required package `spawn-fcgi`

2. Based on https://github.com/sympa-community/sympa/blob/sympa-6.2/service/wwsympa-spawn-fcgi.servicein I created the file `/lib/systemd/system/wwsympa.service` with the following contents:

```
[Unit]
Description=WWSympa - Web interface for Sympa mailing list manager
After=syslog.target sympa.service

[Service]
Type=forking
PIDFile=/run/sympa/wwsympa.pid
ExecStart=/usr/bin/spawn-fcgi -F $FCGI_CHILDREN \
    -P /run/sympa/wwsympa.pid \
    -s /run/sympa/wwsympa.socket \
    -u $FCGI_USER -g $FCGI_GROUP $FCGI_OPTS -- \
    /usr/lib/cgi-bin/sympa/wwsympa.fcgi
Environment="FCGI_CHILDREN=5"
Environment="FCGI_USER=sympa"
Environment="FCGI_GROUP=sympa"
Environment="FCGI_OPTS=-M 0600 -U www-data"
EnvironmentFile=-/etc/sysconfig/sympa
Restart=always

[Install]
WantedBy=multi-user.target
```

3. Based on https://github.com/sympa-community/sympa/blob/sympa-6.2/service/wwsympa-multiwatch.socketin I created the file `/lib/systemd/system/wwsympa.socket` with the following contents:

```
[Unit]
Description=Sympa web interface socket

[Socket]
SocketUser=www-data
SocketMode=0600
ListenStream=/run/sympa/wwsympa.socket

[Install]
WantedBy=sockets.target
```
4. Then I enabled the service with these commands. Be sure that none of them comes out with an error.

```
systemctl daemon-reload
systemctl start wwsympa.service
systemctl status wwsympa.service
systemctl enable wwsympa.service
```
5. Out of somewhere (I cannot recall) I created the file `/etc/apache2/sites-available/atic-sympa.conf` with the following contents:

```
<Location /wws>
	SetHandler "proxy:unix:/run/sympa/wwsympa.socket|fcgi://"
	# Don't forget to edit lines below!
	#Require local
	Require all granted
</Location>

<Location /static-sympa>
	# Don't forget to edit lines below!
	#Require local
	Require all granted
</Location>
Alias /static-sympa /usr/share/sympa/static_content

<Location /css-sympa>
	Require all granted
</Location>
Alias /css-sympa /var/lib/sympa/css
```
6. And finally I enabled the apache required modules and activated the newly created site: 

```
a2enmod proxy_fcgi
a2enmod alias
a2enmod proxy
a2enconf atic-sympa
systemctl restart apache2
```

Thank you very much to everyone.
Best regards.

**Comment by ikedas** [comment]
@qosobrin , you mix two methods that are different each other: Using spawn-fcgi and using Systemd socket.  Please check [the manual](https://sympa-community.github.io/manual/install/configure-http-server.html) to choose only one of them.  (I added cautions not to mix them.)

**Comment by qosobrin** [comment]
You are right, @ikedas. In my simple guide step 3 should be ignored. The rest are adapted from the sections about apache2 in [Using separate FastCGI service](https://sympa-community.github.io/manual/install/configure-http-server-spawnfcgi.html) and [Setup HTTP server as FastCGI proxy](https://sympa-community.github.io/manual/install/configure-http-server-proxy.html) If I'm right, please let me know so I can edit my previous entry for future reference to other people consulting this issue. 

Thank you very much.
Best regards.

**Comment by ikedas** [comment]
Thank you @qosobrin .  Please edit it.
